utilities/constants.ts

```diff
@@ -0,0 +1,4 @@
+export const APPLICATION_JSON = 'application/json'
```

Comment:
`APPLICATION_JSON` might be overkill. This isn't "magic" -- it reads exactly what it is. For example, is 3.14 just 3.14 or is it meant to be `PI`? Is 7 just a number or is it `MAXIMUM_NUMBER_OF_ITEMS`? Is `#3cb371` `MEDIUM_SEA_GREEN` or `MEDIUM_AQUA_MARINE`? Without the constant, you don't know.

Another use is to avoid typos. But both `application/json` and `POST` probably don't need constants. @reubenberghan , thoughts?

Reply:
OK

Reply:
Actually, this is a good candidate for an [Enum](https://www.typescriptlang.org/docs/handbook/enums.html), which TypeScript provides:

```typescript
export enum HttpRequestMethod {
  Delete = 'DELETE'
  Get = 'GET'
  Post = 'POST'
  Put = 'PUT'
}
```

Then you can use the Enum as the type and to assign values:

```typescript
const method: HttpRequestMethod = HttpRequestMethod.Post
```

Very cool.

Reply:
Tried this two way today. They are both enlightening that I haven't thought. In my humble opinion:
* literal type is easier for parameters;
* enum is easier for arguments. 
Please correct me if I missed something.

Reply:
```ts
test('declare a litral type of an argument', () => {
  // method is defined as a string in the real fetch()
  type Option = { method?: string }
  function mockFetch(options: Option): void {
    console.log(options.method)
  }
  type HttpRequestMethods = 'GET' | 'POST'

  const method: HttpRequestMethods = 'GUESS' // Type Error
  mockFetch({ method })
  // Type assertion doesn't work, since HttpRequestMethods overlapped with string
  mockFetch({ method: 'GUESS' as HttpRequestMethods }) // No Error
  mockFetch({ method: <HttpRequestMethods>'GUESS' }) // No Error
})

test('declare a enum type of an argument', () => {
  // method is defined as a string in the real fetch()
  type Option = { method?: string }
  function mockFetch(options: Option): void {
    console.log(options.method)
  }

  enum HttpRequestMethod {
    GET = 'GET',
    POST = 'POST',
  }
  mockFetch({ method: HttpRequestMethod.GET })
  mockFetch({ method: 'GET' })
  mockFetch({ method: 'GUESS' }) // No Error
})

test('declare a litral type for a parameter', () => {
  type HttpRequestMethods = 'GET' | 'POST'
  type Option = { method: HttpRequestMethods }
  function myFetch(options: Option): void {
    console.log(options.method)
  }

  myFetch({ method: 'GUESS' }) // Type Error
})

test('declare a enum type for a parameter', () => {
  enum HttpRequestMethod {
    GET = 'GET',
    POST = 'POST',
  }
  type Option = { method: HttpRequestMethod }
  function myFetch(options: Option): void {
    console.log(options.method)
  }

  myFetch({ method: HttpRequestMethod.GET })
  myFetch({ method: 'GET' }) // Type Error
  myFetch({ method: 'GUESS' }) // Type Error
})
```

Reply:
Let's group types at the top of the test. Blank line before consts. Then blank line before expects:

(I know this is just to show options, but let's get in the habit of grouping like items and using single blank lines to separate groups.)

```typescript
test('declare a litral type of an argument', () => {
  type Option = { method?: string }
  type HttpRequestMethods = 'GET' | 'POST'

  function mockFetch(options: Option): void {
    console.log(options.method)
  }

  const method: HttpRequestMethods = 'GUESS' // Type Error

  mockFetch({ method })

  // Type assertion doesn't work, since HttpRequestMethods overlapped with string
  mockFetch({ method: 'GUESS' as HttpRequestMethods }) // No Error
  mockFetch({ method: <HttpRequestMethods>'GUESS' }) // No Error
})
```